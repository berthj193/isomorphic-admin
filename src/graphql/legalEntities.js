import gql from 'graphql-tag';

export const getList = gql`
query {
  legal_entities {
    id
    name
    stores {
      id
      key
    }
    vat_number
    created_at
  }
}`;

export const getAutocompleteList = gql`
query {
  legal_entities {
    id
    name
  }
}`;

export const createLegalEntity = gql`mutation createLegalEntity(
  $name: String!, $vat_number: String
) {
  createLegalEntity(
    name: $name,
    vat_number: $vat_number
  ) {
    id
  }
}`;

export const destroyLegalEntity = gql`mutation destroyLegalEntity(
  $id: ID!
) {
  destroyLegalEntity(
    id: $id
  )
}`;

export const updateLegalEntity = gql`mutation updateLegalEntity(
  $id: ID!, $name: String, $vat_number: String
) {
  updateLegalEntity(
    id: $id,
    name: $name,
    vat_number: $vat_number
  ) {
    id
  }
}`;
