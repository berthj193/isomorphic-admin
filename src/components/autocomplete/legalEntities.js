import React from 'react';
import PropTypes from 'prop-types';
import withStoresManager from '../apolloProviders/withStoresManager';
import { getAutocompleteList as getLegalEntitiesAutocompleteList } from '../../graphql/legalEntities';
import Select, { SelectOption } from '../uielements/select';

export class LegalEntitiesAutocomplete extends React.Component {
  static propTypes = {
    legal_entities: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      name: PropTypes.string,
    })),
  };

  static defaultProps = {
    legal_entities: [],
  };

  dataSource = (this.props.legal_entities || []).map(({ id, name }) =>
    <SelectOption value={id} key={id}>{`${name}`}</SelectOption>);

  render() {
    return (
      <Select
        showSearch
        {...this.props}
      >
        {this.dataSource}
      </Select>
    );
  }
}

export default withStoresManager(LegalEntitiesAutocomplete, [
  {
    query: getLegalEntitiesAutocompleteList,
  },
]);
