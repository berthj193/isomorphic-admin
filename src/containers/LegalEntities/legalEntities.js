import React, { Component } from 'react';
import PropTypes from 'prop-types';
import columns from './columns';
import { getList, destroyLegalEntity } from '../../graphql/legalEntities';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import ItemList from '../../components/list/item-list';
import DeleteButton from '../../components/uielements/deleteButton';

export class LegalEntities extends Component {
  static propTypes = {
    legal_entities: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        name: PropTypes.string,
        vat_number: PropTypes.string,
        stores: PropTypes.arrayOf(PropTypes.object),
        created_at: PropTypes.string,
      }),
    ),
    error: PropTypes.string,
    mutate: PropTypes.func,
  };

  static defaultProps = {
    legal_entities: [],
  };

  dataSource = () => this.props.legal_entities.map(
    legalEntity => ({
      key: legalEntity.id,
      id: legalEntity.id,
      name: legalEntity.name,
      vat_number: legalEntity.vat_number,
      created_at: legalEntity.created_at,
      stores: legalEntity.stores.length,
    })
  );

  deleteEntity = id => e => {
    e.preventDefault();
    e.stopPropagation();
    this.props.mutate({
      mutation: destroyLegalEntity,
      variables: {
        id,
      },
      messages: ['Legal entity removed successfully'],
    });
  };

  actionColumn = ({ id }) => (
    <ul className="table-action">
      <li>
        <DeleteButton
          onConfirm={this.deleteEntity(id)}
          tooltipText="Remove a legal entity"
        />
      </li>
    </ul>
  );

  columns = [
    ...columns,
    {
      title: 'Action',
      dataIndex: '',
      key: 'a',
      render: this.actionColumn,
    },
  ];

  render() {
    return (
      <ItemList
        data={this.dataSource()}
        error={this.props.error}
        title="Legal entities"
        columns={this.columns}
        buttons={[{
          title: 'New legal entity',
          link: '/dashboard/legal_entities/new',
        }]}
        loading={!this.props.legal_entities && !this.props.error}
      />
    );
  }
}

export default withStoresManager(LegalEntities, { query: getList });
