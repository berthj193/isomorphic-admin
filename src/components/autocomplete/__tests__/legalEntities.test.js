import React from 'react';
import { LegalEntitiesAutocomplete } from '../legalEntities';
import { shallow } from 'enzyme';

describe('legal entities autocomplete component', () => {
  const getComponent = (props = {}) => shallow(<LegalEntitiesAutocomplete {...props} />);
  const dataObj = {
    id: '1',
    name: 'foo',
  };

  it('datasource should return empty array when there are no elements', () => {
    const instance = getComponent().instance();
    expect(instance.dataSource.length).toBe(0);
  });

  it('datasource should return one element', () => {
    const instance = getComponent({
      legal_entities: [dataObj],
    }).instance();

    expect(instance.dataSource).toMatchSnapshot();
  });
});
