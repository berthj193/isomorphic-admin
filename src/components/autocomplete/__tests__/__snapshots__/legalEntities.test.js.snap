// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`legal entities autocomplete component datasource should return one element 1`] = `
Array [
  <Option
    value="1"
>
    foo
</Option>,
]
`;
